### PR TITLE
Add addPolicyName/removePolicyName methods for TrustedTypesDirective

### PR DESCRIPTION
**Pull request by sideshowbarker**
## Summary
- Port the remaining manipulation API from https://github.com/shapesecurity/salvation/pull/273 that was not included in 34b5635
- Add `ManipulationErrorConsumer` interface and `removeValueExact()` to `Directive` base class
- Add `addPolicyName()` and `removePolicyName()` to `TrustedTypesDirective`
- Add 8 new tests covering validation, duplicate warnings, case sensitivity, special characters, round-trip serialization, and edge cases

### CODE REVIEW

File: src/main/java/org/htmlunit/csp/directive/TrustedTypesDirective.java [hunk 1]
```diff
@@ -188,4 +188,32 @@ public void setStar_(final boolean star) {
...
+            throw new IllegalArgumentException("Invalid policy name: " + policyName);
+        }
+        // Policy names are case-sensitive per browser behavior
+        if (policyNames_.contains(policyName)) {
+            errors.add(ManipulationErrorConsumer.Severity.Warning, "Duplicate policy name " + policyName);
+            return;
+        }
+        policyNames_.add(policyName);
+        addValue(policyName);
+    }
```
Comment (kingthorin) [inline src/main/java/org/htmlunit/csp/directive/TrustedTypesDirective.java]: Doesn't this need to handle the allow-duplicates directive?

**Comment by kingthorin** [comment]
I'm not against these methods but if I recall correctly all the manipulation functionality was removed when the library was brought into this repo.

**Comment by sideshowbarker** [comment]
> I'm not against these methods but if I recall correctly all the manipulation functionality was removed when the library was brought into this repo.

Ah OK — well, if the manipulation functionality was intentionally removed, then I don’t feel strongly about adding it back. And in that case, we could just close this unmerged. I don’t need that functionality for the HTML checker, and I don’t personally have any other use cases for it. I had raised this only in a kind of due-diligence spirit of having the patch check off all the boxes only being complete as I could make it. But it seems like we don’t need for it be quite so “complete”…

**Comment by rbri** [comment]
> but if I recall correctly all the manipulation functionality was removed when the library was brought into this repo.

The readme also states this
> For HtmlUnit, the code has been adapted to the code style rules used, and support for editing policies has been removed.

Removing the edit part makes maintenance a bit simpler and as long as noone needs this I like to stay with this.

❤️ cool to have you on board @sideshowbarker, thanks!
